### ISSUE TITLE
Improved handling of object segmentation for better accuracy when parts of a segment fall outside image boundaries.

### ISSUE THREAD

**Issue by corpse-dog** [ultralytics/ultralytics#23596]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

Improved handling of object segmentation for better accuracy when parts of a segment fall outside image boundaries. 
After RandomPerspective, the segment points are set to the 0 axis, but there are actually no pixels to be detected there. I think this will affect the regression of bboxloss and the learning of segments.
original image
<img width="647" height="673" alt="Image" src="https://github.com/user-attachments/assets/3ffb7c49-e1ed-4cbb-b0fd-2d39c9edaa8a" />

RandomPerspective

<img width="642" height="673" alt="Image" src="https://github.com/user-attachments/assets/cd813ed2-2286-422f-8b63-65bf3659d1f4" />

Improved results

<img width="641" height="670" alt="Image" src="https://github.com/user-attachments/assets/f36f33ba-d2e1-4c53-877f-16a7bf807723" />

The solution only needs to modify the inside=(x>=0)&(y>=0)&(x<=width)&(y<=height) in the segment2box function of ops. py to inside=(x>0)&(y>0)&(x<width)&(y<height)

### Environment

Ultralytics 8.3.253 🚀 Python-3.8.10 torch-2.3.0+cu118 CUDA:0 (NVIDIA RTXA6000, 24576MiB)
Setup complete ✅ (16 CPUs, 64.0 GB RAM, 449.6/449.3 GB disk)

OS Windows-10-10.0.1945-SP0
Environment Windows
Python 3.8.10
Install pip
RAM 64.00 GB
Disk  449.6/449.3 GB
CPU Intel Xeon Silver 4216 2.10GHz
CPU count 16
GPU NVIDIA RTXA6000, 24576MiB
GPU 1
CUDA 11.8

### Minimal Reproducible Example

def segment2box(segment, width: int = 640, height: int = 640):
    """Convert segment coordinates to bounding box coordinates.

    Converts a single segment label to a box label by finding the minimum and maximum x and y coordinates. Applies
    inside-image constraint and clips coordinates when necessary.

    Args:
        segment (torch.Tensor): Segment coordinates in format (N, 2) where N is number of points.
        width (int): Width of the image in pixels.
        height (int): Height of the image in pixels.

    Returns:
        (np.ndarray): Bounding box coordinates in xyxy format [x1, y1, x2, y2].
    """
    x, y = segment.T  # segment xy
    # Clip coordinates if 3 out of 4 sides are outside the image
    if np.array([x.min() < 0, y.min() < 0, x.max() > width, y.max() > height]).sum() >= 3:
        x = x.clip(0, width)
        y = y.clip(0, height)
    # inside = (x >= 0) & (y >= 0) & (x <= width) & (y <= height)
    inside = (x > 0) & (y > 0) & (x < width) & (y < height)
    x = x[inside]
    y = y[inside]
    return (
        np.array([x.min(), y.min(), x.max(), y.max()], dtype=segment.dtype)
        if any(x)
        else np.zeros(4, dtype=segment.dtype)
    )  # xyxy

### Additional

The solution only needs to modify the inside=(x>=0)&(y>=0)&(x<=width)&(y<=height) in the segment2box function of ops. py to inside=(x>0)&(y>0)&(x<width)&(y<height)

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @corpse-dog, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

This is an automated response 🤖—an Ultralytics engineer will also assist soon. Thanks for the clear write-up, screenshots, and proposed change 🙏🧩

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. You included a snippet—could you also add an end-to-end MRE that reproduces the behavior from a clean install, including:
-Exact command used to train (CLI or Python)
-A tiny sample label/segment (or a minimal synthetic segment) that triggers the boundary condition
-What output you observe vs. what you expect (e.g., the resulting box/segment after RandomPerspective)
-Any relevant logs and your `ultralytics` version info (you already shared environment details—thank you!)

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Can you open a PR?
